### DISCUSSION TITLE
Add VM labels support

### DISCUSSION THREAD

**Discussion by jay7x** [lima-vm/lima#1309] [Ideas]
### Description

It'd be nice to have VM labels implemented. This way one can filter VM list by labels. So far I'd say it's enough to:
1. Add `labels` list to YAML schema.
2. Attach those labels to a VM internal structure.
3. Display those labels comma-separated in the "LABELS" field of the `limactl list` output (can even be hidden by default, so one can use `--format {{.Labels}}` to display them.

Later some filters can be implemented in the `list`(`start`&`stop`?) command but not really necessary I'd say.. as long as those fields are available as an array in the JSON output one can use it instead.

I'd personally want this to be able to group my VMs without polluting the VM name with group name. Then I can improve my Puppet Bolt inventory plugin so it can auto-assign VMs to groups.

**Comment by jandubois** [comment]
> It'd be nice to have VM labels implemented. This way one can filter VM list by labels.

I don't really understand the use-case here, and think this is too specialized to warrant explicit support in the schema.

You can already emulate this using environment variables, with the bonus feature of making the label available inside the VM:

```console
$ tail -2 ~/.lima/alpine/lima.yaml
env:
  LABEL: foo

$ limactl ls -f '{{.Name}} - {{.Config.Env.LABEL}}'
alpine - foo

$ limactl shell alpine sh -c 'echo $LABEL'
foo
```

**Comment by jay7x** [comment]
Hi @jandubois! Thank you for the workaround! I can use it personally but I wouldn't call this general solution.. but if you don't see a value of labels support I'd ask you to move this issue to discussions page. We can wait for more votes then :)

**Comment by AkihiroSuda** [comment]
Linking the current discussion (covers disks and networks in addition to VM):
- https://github.com/lima-vm/lima/issues/3240
